### PR TITLE
fix register/sign form input red

### PR DESCRIPTION
**Pull request by DanielPalafox**
https://trello.com/c/FIqHqhej/6966-qa-clicking-register-now-on-signin-makes-username-field-red

### CODE REVIEW

**Comment by DanielPalafox** [comment]
[[QA] clicking "register now" on /signin makes username field red](https://trello.com/c/FIqHqhej/6966-qa-clicking-register-now-on-signin-makes-username-field-red)

File: src/app/sign-in/components/form-sign-in/form-sign-in.component.ts [hunk 1]
```diff
@@ -92,26 +91,29 @@ export class FormSignInComponent implements OnInit, AfterViewInit {
...
 
   ngAfterViewInit(): void {
     this.firstInput.nativeElement.focus()
   }
 
   onSubmit() {
-    this.signInLocal.data = this.authorizationForm.getRawValue()
-
-    this.authorizationForm.markAllAsTouched()
+    this.addUsernameValidation()
```
Comment (leomendoza123) [inline src/app/sign-in/components/form-sign-in/form-sign-in.component.ts:113]: ** observation no changes require ** 

Addin the validation after submit is a intersting approach, I think I never see before. 
I believe it should work just fine.

an alternative to this is to write our own ErroStateMatcher as we have on the register. https://github.com/ORCID/orcid-angular/blob/master/src/app/register/ErrorStateMatcherForFormLevelErrors.ts

If we ever have any issue with this approach we can do try that.

**Review by leomendoza123 — CHANGES_REQUESTED** [review]

File: src/app/shared/validators/username/username.validator.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ export class UsernameValidator {
     const orcidError = Validators.pattern(ORCID_REGEXP)(control)
     const orcidUriError = Validators.pattern(ORCID_URI_REGEXP)(control)
 
-    return control.value.length > 19
+    return control.value && control.value.length > 19
```
[the remark this answers is not in the document]
Comment (DanielPalafox) [reply src/app/shared/validators/username/username.validator.ts]: Yes, you are right we should return something immediately there is no case to call `validateUsername` if it is null but we already have a validator for that which is `Validators.required`. So there is no case to return `invalidUsername`. I will do the fix

File: src/app/sign-in/components/form-sign-in/form-sign-in.component.ts (hunk 1 above)
Comment (DanielPalafox) [reply src/app/sign-in/components/form-sign-in/form-sign-in.component.ts:109]: Ohh I see. Ok ok

**Review by leomendoza123 — APPROVED** [review]

**Pull request merged**